### PR TITLE
SV-COMP: Adapt parsing to Task-Definition Format Version 2.0

### PR DESCRIPTION
**Pull request by RyanGlScott**
The `.yml` files included in https://github.com/sosy-lab/sv-benchmarks have undergone some changes since SV-COMP 2020. In particular, they now adhere to `format_version: 2.0`, as described [here](https://github.com/sosy-lab/sv-benchmarks/tree/99d37c5b4072891803b9e5c154127c912477f705#task-definitions). This means that the benchmark sets' descriptions and architectures are no longer described in `.cfg` files, but instead they are included with each `.yml` task file as `options`.

This changes the parsing logic in `Crux.SVCOMP` accordingly.

### CODE REVIEW

File: crux/src/Crux/SVCOMP.hs [hunk 1]
```diff
@@ -186,7 +167,7 @@ loadVerificationTask fp =
  decodeVT :: Yaml.Value -> IO VerificationTask
  decodeVT (Yaml.Object o) =
   do case HM.lookup "format_version" o of
-       Just (Yaml.String "1.0") -> return ()
+       Just (Yaml.String "2.0") -> return ()
        _ -> fail $ unwords ["Expected verification task version 1.0 while parsing", show fp]
```
Comment (robdockins) [inline crux/src/Crux/SVCOMP.hs:171]: should update this error message

File: crux/src/Crux/SVCOMP.hs [hunk 2]
```diff
@@ -280,4 +275,4 @@ deduplicateTasks :: [BenchmarkSet] -> TaskMap
 deduplicateTasks = Map.unionsWith (++) . map f
   where
   f bs = Map.fromList
-           [ ((t, benchmarkArchWidth bs), [bs]) | t <- benchmarkTasks bs ]
+           [ (t, [bs]) | t <- benchmarkTasks bs ]
```
Comment (robdockins) [inline crux/src/Crux/SVCOMP.hs:278]: Should update the comment to this function as well; target architecture is not longer a separate component of the task map.

**Review by robdockins — APPROVED** [review]
Looks good, modulo a couple of string and comments that need updating.

File: crux/src/Crux/SVCOMP.hs (hunk 1 above)
Comment (RyanGlScott) [reply crux/src/Crux/SVCOMP.hs]: Good catch!

File: crux/src/Crux/SVCOMP.hs (hunk 2 above)
Comment (RyanGlScott) [reply crux/src/Crux/SVCOMP.hs:278]: Right you are. Fixed.

**Pull request merged**